### ISSUE TITLE
A question about the implementation detail

### ISSUE THREAD

**Issue by speedcell4** [Jacob-Zhou/FeatureCRFAE#1]
Hi, thanks for sharing the source code.

In your paper Equation 11, I wonder why there is a `LayerNorm`. 
I understand the $s(x, y_i)$ is an unary score function, so I guess it should be a `softmax`?

I am looking forward to hearing back from you.

**Comment by Jacob-Zhou** [comment]
Thank you for your interest in our work.

The use of **LayerNorm** in Equation 11 is indeed correct.
The unary score $s(x, y_i)$ is not necessarily a probability, since it will be globally normalized later by the Forward algorithm.
To be honest, our decision to use the ''LayerNorm'' function in Equation 11 is based more on empirical evidence than theoretical reasoning.
Our preliminary experiments showed that the performance was improved after using the **LayerNorm** function in Equation 11.
However, due to the limitations of our mathematical expertise, we cannot provide a rigorous mathematical proof for it.
I guess that the **LayerNorm** function scales the unary score $s(x, y_i)$ to a proper range, which makes the gradient descent more stable.

I hope this answer addresses your question.

**Comment by speedcell4** [comment]
Wow, I got it. Thank you for your quick response.

**Comment by speedcell4** [comment]
I have tried some similar things on my own project, I guess if the motivation is to limit the unary score in a proper range, then $\tanh$ may be more suitable. Not sure if it is also good for your project, are you interested in having a try on it?

**Comment by Jacob-Zhou** [comment]
Thank you for your advice! I indeed interested in trying the $\mathop{tanh}$ activation function. However, I have shifted my research to a different direction and am quite busy recently. Therefore, it might take a while before I can try it out :(

**Comment by speedcell4** [comment]
Please let me know the results when you have it done.
Thanks in advance.
